percy/environment.py

```diff
@@ -73,43 +73,53 @@ def target_branch(self):
 
     @property
     def commit_data(self):
-        output = ''
-        if self.commit_sha:
-          output = self._raw_commit_output(self.commit_sha)
-        if not output:
-          output = self._raw_commit_output('HEAD')
-
-        # If _raw_commit_output can't be read, return the branch only.
-        if not output:
-          return { 'branch': self.branch }
+        # Try getting commit data from git
+        git_data_from_git = self._raw_commit_output(self.commit_sha or 'HEAD')
```

Comment:
`self.commit_sha or 'HEAD'` is logically different.  It won't retry the command using `HEAD` if trying the command for `self.commit_sha` fails.

Reply:
Ok, good point.

Reply:
Actually, wouldn't it evaluate the argument expression: `self.commit_sha or 'HEAD'` before trying to run `_raw_commit_output`? In which case, it would return None from `self.commit_sha` and then fall back to `HEAD` before calling `_raw_commit_output`?

Reply:
@cadeParade Yes.  But we want also want to run _raw_commit_output with a value for self.commit_sha, not get a result from git, and then try it with `HEAD`.  That was the logic in the current implementation anyway.  Maybe that's not good logic.

Reply:
I wonder if it's better not to have any data from git if there is a value in self.commit_sha, but no commit data available for that SHA.  That's plausible..